files/image_config/config-topology/config-topology.service

```diff
@@ -0,0 +1,19 @@
+[Unit]
+Description=Platform topology configuration service
+After=database.service
+After=database-chassis.service
+Requires=database.service
+Requires=database-chassis.service
```

Comment:
Do we need After and Require dependency to database-chassis?
database service will start only after database-chassis service, so the ordering should automatically be taken care of without this explicit dependency, right?

Reply:
Requires is required for both as the service needs to restart on database or database-chassis restart. It depends on both.
Regarding "After=", defining explicit dependency on order should be fine.

Reply:
without explicit dependency to database-chassis, if database-chassis restarts, database will restart and in turn config-topology should restart. correct?

Reply:
database service unit file does not define "Requires" depenedency on database-chassis. Database service does not restart when database-chassis restarts.

Here are the logs where I restart database-chassis couple of mins ago, but database service did not restart:
root@sonic:/home/cisco# systemctl status database-chassis.service 
● database-chassis.service - database-chassis container
     Loaded: loaded (/lib/systemd/system/database-chassis.service; enabled-runtime; vendor preset: enabled)
     Active: active (running) since Thu 2023-01-19 23:42:30 UTC; 1min 44s ago
   Main PID: 2766315 (database.sh)
      Tasks: 12 (limit: 38080)
     Memory: 23.5M
     CGroup: /system.slice/database-chassis.service
             ├─2766315 /bin/bash /usr/bin/database.sh wait chassisdb
             └─2766324 docker wait database-chassis

Jan 19 23:42:26 sonic systemd[1]: Starting database-chassis container...
Jan 19 23:42:27 sonic database.sh[2766041]: Starting existing database-chassis container
Jan 19 23:42:28 sonic database.sh[2766093]: database-chassis
Jan 19 23:42:29 sonic database.sh[2766215]: Invalid database name input : 'CHASSIS_APP_DB'
Jan 19 23:42:29 sonic database.sh[2766215]: Unable to connect to redis: Cannot assign requested address
Jan 19 23:42:30 sonic systemd[1]: Started database-chassis container.
root@sonic:/home/cisco# systemctl status database
● database.service - Database container
     Loaded: loaded (/lib/systemd/system/database.service; enabled; vendor preset: enabled)
    Drop-In: /etc/systemd/system/database.service.d
             └─auto_restart.conf
     Active: active (running) since Sat 2023-01-14 15:51:46 UTC; 5 days ago
   Main PID: 3127 (database.sh)
      Tasks: 13 (limit: 38080)
     Memory: 32.1M
     CGroup: /system.slice/database.service
             ├─3127 /bin/bash /usr/bin/database.sh wait
             └─3201 docker wait database

Reply:
Thank you for the details!